wdl-analysis/src/handlers/goto_definition.rs

```diff
@@ -0,0 +1,604 @@
+            return Err(anyhow!(
+                "document analysis data not available for {}",
+                document_uri
+            ));
+        };
+
+        let offset = position_to_offset(&lines, position, encoding)?;
+        let Some(token) = find_identifier_token_at_offset(&root, offset) else {
+            return Err(anyhow!("no identifier found at position"));
+        };
+
+        let ident_text = token.text();
+        let parent_node = token
+            .parent()
+            .ok_or_else(|| anyhow!("identifier token has no parent"))?;
+
+        // Context based resolution
+        if let Some(location) = self.resolve_by_context(
+            &parent_node,
+            &token,
+            analysis_doc,
+            &document_uri,
+            &lines,
+            graph,
+        )? {
+            return Ok(Some(location));
+        }
+
+        // Scope based resolution
+        if let Some(scope_ref) = analysis_doc.find_scope_by_position(token.span().start()) {
+            if let Some(name_def) = scope_ref.lookup(ident_text) {
+                return Ok(Some(location(&document_uri, name_def.span(), &lines)?));
+            }
+        }
+
+        // Global resolution
+        self.resolve_global_identifier(analysis_doc, ident_text, &document_uri, &lines, graph)
+    }
+
+    /// Resolves identifier definition based on their parent node's syntax kind
+    fn resolve_by_context(
+        &self,
+        parent_node: &SyntaxNode,
+        token: &SyntaxToken,
+        analysis_doc: &Document,
+        document_uri: &Url,
+        lines: &Arc<LineIndex>,
+        graph: &DocumentGraph,
+    ) -> Result<Option<GotoDefinitionResponse>> {
+        match parent_node.kind() {
+            SyntaxKind::TypeRefNode => {
+                self.resolve_type_reference(analysis_doc, token.text(), document_uri, lines, graph)
+            }
+
+            SyntaxKind::CallTargetNode => self.resolve_call_target(
+                parent_node,
+                token,
+                analysis_doc,
+                document_uri,
+                lines,
+                graph,
+            ),
+            SyntaxKind::ImportStatementNode => {
+                self.resolve_import_namespace(parent_node, token, document_uri, lines)
+            }
+
+            SyntaxKind::AccessExprNode => self.resolve_access_expression(
+                parent_node,
+                token,
+                analysis_doc,
+                document_uri,
+                lines,
+                graph,
+            ),
+
+            // handled by scope resolution
+            SyntaxKind::NameRefExprNode => Ok(None),
+            _ => Ok(None),
+        }
+    }
+
+    /// Resolves type references to their definition locations.
+    ///
+    /// Searches for struct definitions in the current document and imported
+    /// namespaces
+    fn resolve_type_reference(
+        &self,
+        analysis_doc: &Document,
+        ident_text: &str,
```

Comment:
Can we just call this `token` to stay consistent with the other methods and call `.text()` later on when it is needed?